scalyr_agent/metrics/functions.py

```diff
@@ -407,6 +433,42 @@ def should_calculate(cls, monitor, metric_name, extra_fields=None):
 
         return False
 
+    @classmethod
+    def _remove_old_entries(cls, monitor):
+        # type: (ScalyrMonitor) -> None
+        """
+        This function removes old entries for values with the timestamps which are older than the
+        defined threshold.
+
+        TODO: Thread safety?
```

Comment:
Yeah.. I guess we could guard it by some singleton lock but I'am not sure how it may affect performance..

Reply:
For now I avoided it since ``calculate()`` is called on every single ``emit_value()`` function call and it could add massive overhead.

I plan to look more into thread safety and performance in the future, perhaps by moving some of the abstractions directly on ``ScalyrMonitor`` to mitigate thread safety and performance issues (since ``RateMetricFunction()`` class instance would then be monitor specific and monitors already run in a separate dedicated thread).